File: sdk/highlight-java/highlight-sdk/src/main/java/io/highlight/sdk/HighlightRoute.java
package io.highlight.sdk;

/**
 * This class build the route for the log and trace backend URL's
 */
public class HighlightRoute {

	// The default backend URL's.
	public static final String DEFAULT_OTEL_BACKEND = "https://otel.highlight.io:4318";
	public static final String DEFAULT_GRAPHQL_BACKEND = "https://pub.highlight.run";

	// The default routes for sending.
	public static final String ROUTE_LOGS = "v1/logs";
	public static final String ROUTE_TRACES = "v1/traces";

	/**
	 * Builds a route for sending logs to Highlight, based on the provided backend
	 * URL. <br>
	 * If no backend URL is provided, the default backend URL is used.
	 * 
	 * @param backend The backend URL to use.
	 * @return The route for sending logs to Highlight.
	 */
	public static String buildLogRoute(String backend) {
		if (backend == null) {
			return DEFAULT_OTEL_BACKEND + "/" + ROUTE_LOGS;
		}

		if (backend.endsWith(ROUTE_LOGS)) {
			return backend;
		}

		if (backend.endsWith("/")) {
			return backend + ROUTE_LOGS;
		}

		return backend + "/" + ROUTE_LOGS;
	}

	/**
	 * Builds a route for sending traces to Highlight, based on the provided backend
	 * URL. <br>
	 * If no backend URL is provided, the default backend URL is used.
	 * 
	 * @param backend The backend URL to use.
	 * @return The route for sending traces to Highlight.
	 */
	public static String buildTraceRoute(String backend) {
		if (backend == null) {
			return DEFAULT_OTEL_BACKEND + "/" + ROUTE_TRACES;
		}

		if (backend.endsWith(ROUTE_TRACES)) {
			return backend;
		}

		if (backend.endsWith("/")) {
			return backend + ROUTE_TRACES;
		}

		return backend + "/" + ROUTE_TRACES;
	}

	/**
	 * Non accessible constructor because this class just providing static fields
	 */
	private HighlightRoute() {
	}
}

File: sdk/highlight-php/src/SDK/HighlightRoute.php
<?php

namespace Highlight\SDK;

class HighlightRoute {
    // The default backend URL's.
    const DEFAULT_BACKEND = "https://otel.highlight.io:4318";

    // The default routes for sending.
    const ROUTE_LOGS = "v1/logs";
    const ROUTE_TRACES = "v1/traces";

    /**
     * Builds a route for sending logs to Highlight, based on the provided backend
     * URL. If no backend URL is provided, the default backend URL is used.
     *
     * @param string $backend The backend URL to use.
     * @return string The route for sending logs to Highlight.
     */
    public static function buildLogRoute($backend = null) {
        if ($backend === null || trim($backend) === "") {
            return self::DEFAULT_BACKEND . "/" . self::ROUTE_LOGS;
        }

        if (substr($backend, -strlen(self::ROUTE_LOGS)) === self::ROUTE_LOGS) {
            return $backend;
        }

        if (substr($backend, -1) === "/") {
            return $backend . self::ROUTE_LOGS;
        }

        return $backend . "/" . self::ROUTE_LOGS;
    }

    /**
     * Builds a route for sending traces to Highlight, based on the provided backend
     * URL. If no backend URL is provided, the default backend URL is used.
     *
     * @param string $backend The backend URL to use.
     * @return string The route for sending traces to Highlight.
     */
    public static function buildTraceRoute($backend = null) {
        if ($backend === null || trim($backend) === "") {
            return self::DEFAULT_BACKEND . "/" . self::ROUTE_TRACES;
        }

        if (substr($backend, -strlen(self::ROUTE_TRACES)) === self::ROUTE_TRACES) {
            return $backend;
        }

        if (substr($backend, -1) === "/") {
            return $backend . self::ROUTE_TRACES;
        }

        return $backend . "/" . self::ROUTE_TRACES;
    }

    /**
     * Get the default backend URL.
     *
     * @return string The default backend URL.
     */
    public static function getDefaultBackend() {
        return self::DEFAULT_BACKEND;
    }

    // Private constructor to prevent instantiation
    private function __construct() {
    }
}